sqlserver/datadog_checks/sqlserver/utils.py

```diff
@@ -35,9 +41,34 @@ def set_default_driver_conf():
         # Use default `./driver_config/odbcinst.ini` when Agent is running in docker.
         # `freetds` is shipped with the Docker Agent.
         os.environ.setdefault('ODBCSYSINI', DRIVER_CONFIG_DIR)
-    else:
+    elif Platform.is_linux():
+        """
+        The agent running on Linux has msodbcsql18 and FreeTDS installed.
+        The default driver is msodbcsql18.
+        To best leverage the default driver, we set the ODBCSYSINI environment variable to the directory
+        containing the pre-configured odbcinst.ini file.
+        However, if the user has already configured the ODBCSYSINI environment variable,
+        OR if the user has already created or copied the odbcinst.ini file in the unixODBC sysconfig location,
+        we do not override the ODBCSYSINI environment variable.
+        """
+        if pyodbc is None:
+            return
+
+        if 'ODBCSYSINI' in os.environ:
+            # If ODBCSYSINI is already set in env, don't override it
+            return
+
+        if pyodbc.drivers() or pyodbc.dataSources():
+            # If there are already drivers or dataSources installed, don't override the ODBCSYSINI
+            # This means user has copied odbcinst.ini and odbc.ini to the unixODBC sysconfig location
```

Comment:
So you test this part by relying on the fact that we have these files copied on a machine where we conduct tests ? 

Would be may be nice to make sure this statement "
If there are already drivers or dataSources installed, don't override the ODBCSYSINI
This means user has copied odbcinst.ini and odbc.ini to the unixODBC sysconfig location" 
always holds true by physically moving files back and force in the test. this will ensure that any changes to pyodbc wont break our code.

Reply:
https://github.com/DataDog/integrations-core/pull/17729/files#diff-6d9aa168f975bdf165f3cdc4e234dd74949759d4c12b4e357e00cf42eece8c88R446-R450 covers the case where `ODBCSYSINI` is ignored but `odbcinst.ini` are present in the sysconfig

Reply:
`pyodbc.drivers()` checks for driver config and `pyodbc.dataSources()` checks DSN.

`pyodbc` relies on unixODBC to manages the odbc drivers. If unixODBC cannot locate any drivers, `pyodbc.drivers()` will be empty.
The env var `ODBCSYSINI` impacts the unixODBC syspath of both drivers and DSN. Checking `pyodbc.dataSources()` for DSN makes sure the integration acknowledges user configured `odbc.ini` before defaulting the unixODBC syspath. 

I don't see relying on `pyodbc.drivers` and `pyodbc.dataSources` being an issue. In case of a breaking change in the API, it will be caught by tests during pakcage version bump.